### PR TITLE
Pass integration configuration to "extendApp" and fix CT cookies

### PR DESCRIPTION
**Pull request by filipsobol**
### Short Description of the PR
* [BREAKING] Pass object containing `app` and `configuration` to `extendApp` instead of `app` object only.
* Adds  `expires` to CT cookie to the same date as the token, so the cookie is stored for as long as the token is valid.

### Pull Request Checklist
- [X] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [X] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [X] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [X] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

**Review by Baroshem — COMMENTED** [review]
I tried different approach and it did not work for me. If that works for you, then you have my approval ;)

**Comment by filipsobol** [comment]
> I tried different approach and it did not work for me. If that works for you, then you have my approval ;)

It does. The problem is that the cookie is set not only by `onTokenChange`, but also `onTokenRead`.

File: packages/commercetools/api-client/src/index.server.ts [hunk 1]
```diff
@@ -70,13 +70,23 @@ const tokenExtension: ApiClientExtension = {
         auth: {
           onTokenChange: (newToken) => {
             if (!currentToken || currentToken.access_token !== newToken.access_token) {
-              res.cookie('vsf-commercetools-token', JSON.stringify(newToken));
+              res.cookie(
+                'vsf-commercetools-token',
+                JSON.stringify(newToken),
+                { expires: newToken?.expires_at ? new Date(newToken.expires_at) : new Date() }
```
Comment (Fifciu) [inline packages/commercetools/api-client/src/index.server.ts:76]: Maybe?:
```ts
{ expires: new Date(newToken?.expires_at || null) }
```

Comment (filipsobol) [reply packages/commercetools/api-client/src/index.server.ts]: I pushed a fix. Setting date to the past (which will be a case if we do `new Date(null)` removes a cookie, which is not intended behavior.

**Review by Fifciu — APPROVED** [review]

**Review by Baroshem — APPROVED** [review]
Well done!

**Pull request merged**